design/open/gc-on-parquet.md

```diff
@@ -0,0 +1,25 @@
+## Garbage collection using Parquet format
+
+### Motivation
+
+Garbage collection in lakeFS is struggling at large scale.
+
+### Why Parquet?
+
+Parquet is more than a file format. It's a breathing open-source project aiming to provide exactly what we're looking for here: efficient data retrieval. GC is essentially a huge anti-join, and it may be very difficult to achieve a similar optimization in our dedicated format.
+
+
+### Steps
+
+#### 1. Sync
+Ranges of the repository are translated into Parquet files. These are stored in a new metadata location under the repository's storage namespace.
+
+#### 2. Sanity
+Verify all the ranges required for this run are present, and that the number of entries in each range is correct.
+
+#### 3. Start
+Do an anti-join extracting all the addresses that need to be deleted.
+
+### Considerations
```

Comment:
Another thing that comes to mind is that this solution won't be reusable by GC uncommitted, since it is specific to lakeFS metadata processing.

Reply:
Not sure I understand - why is it more/less usable then using a dedicated format?